### PR TITLE
Fix perf tests build

### PR DESCRIPTION
**Pull request by nohwnd**
## Description
Fixing the perf-tests that are in our repo and that are not run.

### CODE REVIEW

**Comment by nohwnd** [comment]
- [ ] They don't run for .NET, I did not look into that yet, but there are bunch of compiler directives, that make the tests just pass for net6.0 (time that any action takes is 0 ms, so it is always less than threshold), I am throwing exception instead
- [ ] The telemetry tests post telemetry, but don't have thresholds, should they? 
- [ ] Other tests are not posting telemetry
- [ ] Is the telemetry useful, and does it have the info that help us split it to scenarios?
- [ ] The telemetry probably should gather some system info to tell us how fast the computer running it is.
- [ ] The thresholds seem arbitrary, some are too low so tests fail. Cannot say if that is because of TP being slow, or because of the threshold being too low.
- [ ] These tests don't run in our CI pipeline.

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs [hunk 1]
```diff
@@ -49,78 +52,76 @@ public class ProtocolV1Tests
...
+    private static string SerializeV1<T>(T data)
     {
-        return JsonDataSerializer.Instance.Deserialize<T>(json, version);
+        return JsonDataSerializer.Instance.Serialize(data, version: 1);
     }
 
-    private static void VerifyPerformanceResult(string scenario, long expectedElapsedTime, long elapsedTime)
+    private static T DeserializeV1<T>(string json)
     {
-        Assert.IsTrue(elapsedTime < expectedElapsedTime, $"Scenario '{scenario}' doesn't match with expected elapsed time.");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs:123]: [minor] You have replaced all calls with <=.

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 2]
```diff
@@ -49,76 +52,80 @@ public class ProtocolV2Tests
...
     {
-        return JsonDataSerializer.Instance.Deserialize<T>(json, version);
+        return JsonDataSerializer.Instance.Deserialize<T>(json, version: 2);
     }
 
     private static void VerifyPerformanceResult(string scenario, long expectedElapsedTime, long elapsedTime)
     {
         Assert.IsTrue(elapsedTime < expectedElapsedTime, $"Scenario '{scenario}' doesn't match with expected elapsed time.");
-        File.AppendAllText(@"E:\ProtocolPerf.txt", $" {scenario} : " + elapsedTime);
+        // File.AppendAllText(@"E:\ProtocolPerf.txt", $" {scenario} : " + elapsedTime);
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:129]: If that's no longer needed, could we remove it?

File: test/Microsoft.TestPlatform.PerformanceTests/PerformanceTests.cs [hunk 3]
```diff
@@ -28,8 +31,7 @@ public void ExecutionPerformanceTest()
         AnalyzePerfData();
         var actualExecutionTime = GetExecutionTime();
 
-        // Sample Assert statement to verify the performance. 500 will be replaced by the actual threshold value.
-        Assert.IsTrue(actualExecutionTime < 500);
+        actualExecutionTime.Should().BeLessOrEqualTo(500.Milliseconds());
```
[the remark this answers is not in the document]
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/PerformanceTests.cs:34]: True. Now the task has 1ms more to run. 😁

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs (hunk 1 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs:123]: yup.

File: scripts/perf/perfconfig.csv [hunk 4]
```diff
@@ -1,4 +1,4 @@
 payload;adapter;runners;discoverygoal;executiongoal
-"MSTestAdapterPerfTestProject"; "MsTest"; ("vstest.console"); (3500); (9500)
-"NUnitAdapterPerfTestProject"; "nUnit"; ("vstest.console","nunit.consolerunner"); (6500, 3500); (45000,40000)
-"XUnitAdapterPerfTestProject"; "xUnit"; ("vstest.console","xunit.runner.console"); (6500, 0); (50000, 38000)
+"MSTest10kPassing"; "MsTest"; ("vstest.console"); (3500); (9500)
+"NUnit10kPassing"; "nUnit"; ("vstest.console","nunit.consolerunner"); (6500, 3500); (45000,40000)
```
Comment (Evangelink) [inline scripts/perf/perfconfig.csv:3]: Nit: Add space after comma.

```suggestion
"NUnit10kPassing"; "nUnit"; ("vstest.console", "nunit.consolerunner"); (6500, 3500); (45000, 40000)
```

File: scripts/perf/perfconfig.csv [hunk 5]
```diff
@@ -1,4 +1,4 @@
...
+"XUnit10kPassing"; "xUnit"; ("vstest.console","xunit.runner.console"); (6500, 0); (50000, 38000)
```
Comment (Evangelink) [inline scripts/perf/perfconfig.csv:4]: Nit: Add space after comma.

```suggestion
"XUnit10kPassing"; "xUnit"; ("vstest.console", "xunit.runner.console"); (6500, 0); (50000, 38000)
```

File: test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj [hunk 6]
```diff
@@ -8,13 +8,12 @@
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
     <OutputType Condition=" '$(TargetFramework)' == 'net6.0' ">Exe</OutputType>
-    <TargetFrameworks>net6.0;net48</TargetFrameworks>
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj:11]: Any reason to remove the `net6.0` build?

File: test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj [hunk 7]
```diff
@@ -8,13 +8,12 @@
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
     <OutputType Condition=" '$(TargetFramework)' == 'net6.0' ">Exe</OutputType>
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj:10]: Looks like this condition is no longer needed (you removed net6.0).

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerfAnalyzer.cs [hunk 8]
```diff
@@ -2,19 +2,18 @@
...
-#if NETFRAMEWORK
 using System;
 using System.Linq;
 
 using Microsoft.Diagnostics.Tracing;
 using Microsoft.Diagnostics.Tracing.Parsers;
 using Microsoft.Diagnostics.Tracing.Session;
-#endif
 
 #nullable disable
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerfAnalyzer.cs:14]: [minor] As you do quite some changes to the class, might be worth to try to remove the `#nullable disable`.

File: test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj (hunk 6 above)
Comment (Evangelink) [reply test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj:11]: Looking at the tests it seems to be removing the specific conditions, right?

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerformanceTestBase.cs [hunk 9]
```diff
@@ -20,7 +23,12 @@ public class PerformanceTestBase : IntegrationTestBase
     public PerformanceTestBase()
         : base()
     {
+#if NET
+
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerformanceTestBase.cs:27]: Nit: Remove blank line

```suggestion        
```

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerformanceTestBase.cs [hunk 10]
```diff
@@ -1,11 +1,14 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Collections.Generic;
 
+using Microsoft.TestPlatform.TestUtilities;
+
 #nullable disable
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerformanceTestBase.cs:9]: [minor] You could also remove this `#nullable disable`.

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerformanceTestBase.cs [hunk 11]
```diff
@@ -37,17 +45,10 @@ public PerformanceTestBase()
...
-        _perfAnalyzer.EnableProvider();
-#endif
-        // Run Test
-        InvokeVsTestForExecution(testAsset, testAdapterPath, ".NETFramework,Version=v4.5.1", runSettings);
-
-        // Stop Listening
-#if NETFRAMEWORK
-        _perfAnalyzer.DisableProvider();
-#endif
+        using(_perfAnalyzer.Start())
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerformanceTestBase.cs:48]: [cosmetic] Nit

```suggestion
        using (_perfAnalyzer.Start())
```

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/TestPlatformTask.cs [hunk 12]
```diff
@@ -27,3 +27,15 @@ public TestPlatformTask(string taskName, double eventStarted)
...
+{
+    public TestPlatformEvent(string eventName, double timeSinceStart)
+    {
+        Name = eventName;
+        TimeSinceStart = timeSinceStart;
+    }
+
+    public string Name { get; }
+    public double TimeSinceStart { get; }
+}
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/TestPlatformTask.cs:41]: Nit: Use a record instead

```suggestion
public record TestPlatformEvent(string Name, double TimeSinceStart);
```

File: test/Microsoft.TestPlatform.PerformanceTests/PerformanceTests.cs [hunk 13]
```diff
@@ -13,23 +17,19 @@ namespace Microsoft.TestPlatform.PerformanceTests;
 /// The performance tests.
 /// </summary>
 [TestClass]
-public class PerformanceTests : PerformanceTestBase
+public class PerformanceTests : TelemetryPerfTestbase
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/PerformanceTests.cs:20]: Nit: Use upper-case for `Base`.

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs [hunk 14]
```diff
@@ -30,97 +32,96 @@ public class ProtocolV1Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Serialize(TestCase);
+            SerializeV1(TestCase);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestCaseSerialize1", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when serializing 10k test cases");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs:65]: Nit: interpolation is not needed.

```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when serializing 10k test cases");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs [hunk 15]
```diff
@@ -30,97 +32,96 @@ public class ProtocolV1Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Deserialize<TestCase>(json);
+            DeserializeV1<TestCase>(json);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestCaseDeserialize1", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when de-serializing 10k test cases");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs:82]: Nit: interpolation is not needed.

```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when de-serializing 10k test cases");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs [hunk 16]
```diff
@@ -30,97 +32,96 @@ public class ProtocolV1Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Serialize(TestResult);
+            SerializeV1(TestResult);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestResultSerialize1", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when serializing 10k test results");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs:98]: Nit: interpolation is not needed.
```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when serializing 10k test results");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs [hunk 17]
```diff
@@ -30,97 +32,96 @@ public class ProtocolV1Tests
...
         sw.Stop();
 
-        VerifyPerformanceResult("TestResultDeserialize1", 3500, sw.ElapsedMilliseconds);
-    }
-
-    private static string Serialize<T>(T data, int version = 1)
-    {
-        return JsonDataSerializer.Instance.Serialize(data, version);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(3.5.Seconds(), $"when de-serializing 10k test results");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV1Tests.cs:115]: Nit: interpolation is not needed.
```suggestion
        actualDuration.Should().BeLessOrEqualTo(3.5.Seconds(), "when de-serializing 10k test results");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 18]
```diff
@@ -49,76 +52,79 @@ public class ProtocolV2Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Serialize(TestCase, 2);
+            SerializeV2(TestCase);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestCaseSerialize2", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when serializing 10k test cases");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:64]: Nit: interpolation is not needed.
```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when serializing 10k test cases");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 19]
```diff
@@ -49,76 +52,79 @@ public class ProtocolV2Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Deserialize<TestCase>(json, 2);
+            DeserializeV2<TestCase>(json);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestCaseDeserialize2", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when de-serializing 10k test cases");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:81]: Nit: interpolation is not needed.
```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when de-serializing 10k test cases");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 20]
```diff
@@ -49,76 +52,79 @@ public class ProtocolV2Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Serialize(TestResult, 2);
+            SerializeV2(TestResult);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestResultSerialize2", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when serializing 10k test results");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:96]: Nit: interpolation is not needed.
```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when serializing 10k test results");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 21]
```diff
@@ -49,76 +52,79 @@ public class ProtocolV2Tests
...
         for (int i = 0; i < 10000; i++)
         {
-            Deserialize<TestResult>(json, 2);
+            DeserializeV2<TestResult>(json);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestResultDeserialize2", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when de-serializing 10k test results");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:113]: Nit: interpolation is not needed.
```suggestion
        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), "when de-serializing 10k test results");
```

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 22]
```diff
@@ -49,76 +52,79 @@ public class ProtocolV2Tests
...
         Stopwatch sw = Stopwatch.StartNew();
         for (int i = 0; i < 10000; i++)
         {
-            Serialize(TestCase, 2);
+            SerializeV2(TestCase);
         }
         sw.Stop();
 
-        VerifyPerformanceResult("TestCaseSerialize2", 2000, sw.ElapsedMilliseconds);
+        var actualDuration = sw.Elapsed;
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:63]: I am fine if you keep it but `sw.Elapsed.Should()` would be clear too.

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs [hunk 23]
```diff
@@ -49,76 +52,79 @@ public class ProtocolV2Tests
...
+        actualDuration.Should().BeLessOrEqualTo(2.Seconds(), $"when serializing 10k test cases");
     }
 
     [TestMethod]
-    public void TestCaseDeserialize2()
+    public void TestCaseDeserializeV22()
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:68]: Could we find a better name than V2 2?

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/EventHandler/RunEventHandler.cs [hunk 24]
```diff
@@ -57,6 +51,7 @@ public void HandleLogMessage(TestMessageLevel level, string message)
         }
 
         Metrics = testRunCompleteArgs.Metrics;
+        LogMessages.Add($"[ERROR] {testRunCompleteArgs.Error}");
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/EventHandler/RunEventHandler.cs:54]: Is it always an error case to come here?

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs [hunk 25]
```diff
@@ -17,26 +17,36 @@
 namespace Microsoft.TestPlatform.PerformanceTests.TranslationLayer;
 
 [TestClass]
-public class TelemetryPerfTestbase
+public class TelemetryPerfTestbase : PerformanceTestBase
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs:20]: Nit: `TelemetryPerfTestBase`.

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs [hunk 26]
```diff
@@ -17,26 +17,36 @@
...
-        var properties = new Dictionary<string, string>();
+        var properties = new Dictionary<string, string>
+        {
+            ["Version"] = "1.0.1",
+            ["Project"] = projectName,
+            ["Scenario"] = scenario,
+            ["Configuration"] = BuildConfiguration,
+        };
+
+
```
Comment (Evangelink) [inline test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs:49]: Nit: un-needed blank line

```suggestion
```

File: test/TestAssets/performance/MSTest1Passing/MSTests.cs [hunk 27]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace MSTestAdapterPerfTests
+{
+    [TestClass]
+	public class MSTests
```
Comment (Evangelink) [inline test/TestAssets/performance/MSTest1Passing/MSTests.cs:7]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/MSTest1Passing/MSTests.tt [hunk 28]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
```
Comment (Evangelink) [inline test/TestAssets/performance/MSTest1Passing/MSTests.tt:1]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/NUnit1000Passing/NunitTests.tt [hunk 29]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace NUnitAdapterPerfTests
+{
+
+	public class NunitTests
```
Comment (Evangelink) [inline test/TestAssets/performance/NUnit1000Passing/NunitTests.tt:7]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/NUnit100Passing/NunitTests.tt [hunk 30]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace NUnitAdapterPerfTests
+{
+
+	public class NunitTests
```
Comment (Evangelink) [inline test/TestAssets/performance/NUnit100Passing/NunitTests.tt:7]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/NUnit1Passing/NunitTests.tt [hunk 31]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace NUnitAdapterPerfTests
+{
+
+	public class NunitTests
```
Comment (Evangelink) [inline test/TestAssets/performance/NUnit1Passing/NunitTests.tt:7]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/XUnit1000Passing/XunitTests.tt [hunk 32]
```diff
@@ -0,0 +1,49 @@
+﻿using Xunit;
+
+namespace XUnitAdapterPerfTests
+{
+	public class XuniTests
```
Comment (Evangelink) [inline test/TestAssets/performance/XUnit1000Passing/XunitTests.tt:5]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/XUnit100Passing/XunitTests.tt [hunk 33]
```diff
@@ -0,0 +1,49 @@
+﻿using Xunit;
+
+namespace XUnitAdapterPerfTests
+{
+	public class XuniTests
```
Comment (Evangelink) [inline test/TestAssets/performance/XUnit100Passing/XunitTests.tt:5]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/TestAssets/performance/XUnit1Passing/XunitTests.tt [hunk 34]
```diff
@@ -0,0 +1,49 @@
+﻿using Xunit;
+
+namespace XUnitAdapterPerfTests
+{
+	public class XuniTests
```
Comment (Evangelink) [inline test/TestAssets/performance/XUnit1Passing/XunitTests.tt:5]: There is a mix of spaces and tabs, could you save the file from VS to have it fixed please?

File: test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj (hunk 6 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj:11]: Yes, I am not very interested in the test "orchestrator" to run twice. Right now a net48 project is running net451 vstest.console, which runs net6.0 tests. The project did not immediately build when having also net6.0 TFM and I don't want to spend time figuring out how I should do event tracing on net6.0 when I have a working solution on net48.

File: test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj (hunk 7 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/Microsoft.TestPlatform.PerformanceTests.csproj:10]: I am pretty sure I would forget to put it back when I make this work on .NET, if ever. So let's just keep it.

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerfAnalyzer.cs (hunk 8 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/PerfAnalyzer.cs:14]: There is only so much you can do in 1 day. ;)

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/EventHandler/RunEventHandler.cs (hunk 24 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/EventHandler/RunEventHandler.cs:54]: nope added condition.

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs (hunk 25 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs:20]: fixed

File: test/TestAssets/performance/MSTest1Passing/MSTests.cs (hunk 27 above)
Comment (nohwnd) [reply test/TestAssets/performance/MSTest1Passing/MSTests.cs:7]: That file is generated by template. Let's see what the template has.

Comment (nohwnd) [reply test/TestAssets/performance/MSTest1Passing/MSTests.cs:7]: Fixed it in the template and replaced everywhere.

File: test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/TestPlatformTask.cs (hunk 12 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/TestPlatformTask.cs:41]: Yeah that would be nice, probably readonly struct record even.

Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/TestPlatformTask.cs:41]: Will skip for now though.

File: test/Microsoft.TestPlatform.PerformanceTests/PerformanceTests.cs (hunk 13 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/PerformanceTests.cs:20]: Fixed.

File: test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs (hunk 23 above)
Comment (nohwnd) [reply test/Microsoft.TestPlatform.PerformanceTests/ProtocolV2Tests.cs:68]: Renamed to just TestCaseDeserialize2, will probably become data driven test later on anyway.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**